pkg/operator/operator.go

```diff
@@ -573,6 +573,22 @@ func (o *Operator) enqueue(obj interface{}) {
 	o.queue.Add(key)
 }
 
+type proxyConfigSupplier func() (*ProxyConfig, error)
+
+func getProxyReader(proxyConfigSupplier proxyConfigSupplier, config *manifests.Config) manifests.ProxyReader {
+	var proxyConfig manifests.ProxyReader = config
+	if proxyConfig.HTTPProxy() == "" && proxyConfig.HTTPSProxy() == "" && proxyConfig.NoProxy() == "" {
```

Comment:
I find it easier to read if we return early in the simple case.

```suggestion
	if proxyConfig.HTTPProxy() != "" || proxyConfig.HTTPSProxy() != "" || proxyConfig.NoProxy() != "" {
		return config
	}
```